File: src/app/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Category } from './category';

@Injectable({
  providedIn: 'root'
})
export class CategoryService {
  baseLink="http://localhost:8080/api/category/v2/"
  constructor(private httpClient:HttpClient) { }

  getAllCategorys():Observable<Category[]>{
    return this.httpClient.get<Category[]>(`${this.baseLink+"getALLCategory"}`);
  }
  addCategory(category :Category):Observable<Category>{
    return this.httpClient.post<Category>(`${this.baseLink+"addCategory"}`,category);
  }
  getCategoryById(id:number):Observable<Category>{
    return this.httpClient.get<Category>(`${this.baseLink+"getCategoryById"}/${id}`);
  }
  assignBrandToCategory(categoryId:number,brandId:number):Observable<Object>{//,brandId:number//categoryId:number
    return this.httpClient.
    put(`${this.baseLink+"assignBrandToCategory"}/${categoryId}/brand/${brandId}`,null);
  }
  deleteCategory(id:number):Observable<Object>{
    return this.httpClient.delete(`${this.baseLink+"deleteCategory/"}${id}`);
  }
  updateCategory(id:number,category:Category):Observable<Object>{
    return this.httpClient.put(`${this.baseLink+"updateCategory/"}${id}`,category)
  }
  getCategoryByName(categoryName:string):Observable<Category>{
    return this.httpClient.get<Category>(`${this.baseLink+"getCategoryByName/"}${categoryName}`);
  }
}

File: src/main/java/com/spring/products/service/CategoryService.java
package com.spring.products.service;

import com.spring.products.entity.Category;

import java.util.List;

public interface CategoryService {
    public Category addCategory(Category category);
    public List<Category> getAllCategory();
    public Category updateCategory(long id,Category category);
    public Category getCategoryById(long id);
    public Boolean deleteCategory(long id);
    public Boolean checkExistenceCategory(String brandName);
    public Category updateBrandOfCategory(long categoryId,long brandId);
    public Category assignBrandToCategory(Long categoryId, Long brandId);
    public Category getCategoryByName(String categoryName);
}
